pkg/icingadb/driver.go

```diff
@@ -1,29 +1,42 @@
-package driver
+package icingadb
 
 import (
 	"context"
-	"database/sql"
 	"database/sql/driver"
 	"github.com/go-sql-driver/mysql"
 	"github.com/icinga/icingadb/pkg/backoff"
 	"github.com/icinga/icingadb/pkg/icingaredis/telemetry"
 	"github.com/icinga/icingadb/pkg/logging"
 	"github.com/icinga/icingadb/pkg/retry"
-	"github.com/jmoiron/sqlx"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"time"
 )
 
-const MySQL = "icingadb-mysql"
-const PostgreSQL = "icingadb-pgsql"
+// Driver names that are automatically registered by the respective packages under exactly these names.
+// Whenever the default names in their packages are altered or should we use other driver packages of these kind,
+// they should always be registered under these names. We do not intend to change these constant names ever again.
+const (
+	MySQL      string = "mysql"
+	PostgreSQL string = "postgres"
+)
```

Comment:
Isn't the comment just saying that the values of the constants are the driver names as registered in the `database/sql` packages by the drivers themselves?

> Whenever the default names in their packages are altered or should we use other driver packages of these kind, they should always be registered under these names.

I'm not exactly sure what "under these names" is supposed to say exactly. But that sentence can be interpreted as "if we would device to use `pgx` instead of `postgres`, we'd still have to make sure it's registered as `postgres`" which sounds strange.

Also, it sounds unlikely that a driver implementation would change the name it registers with `database/sql`, so I don't think it needs to be mentioned explicitly (it should be clear from "as registered with `database/sql`" that if that changes, these constants have to be adapted as well).

> We do not intend to change these constant names ever again.

Why? Especially why does it need to be mentioned here? Sounds a bit like "I know things will break if you change it but I won't tell you why" at the moment.

Reply:
> But that sentence can be interpreted as "if we would device to use `pgx` instead of `postgres`, we'd still have to make sure it's registered as `postgres`" which sounds strange.

Why does this feel strange? After all, no matter which drivers we use, they all have to be either of type `postgres` or `mysql` and it makes no sense to refer to a driver like `pgx` just because its maintainers have named it so. Btw. this sentence just says that if we ever use a new driver that is not yet known by `sqlx`, we need to make sure we register it using one of these constants.